### PR TITLE
bcc: De-vendor libbpf

### PR DESCRIPTION
**Pull request by daniel-thompson**
Currently bcc builds against a vendored copy of libbpf. This causes problems for bpftrace which is built against bcc and the system libbpf. The resulting version skew between the vendored and system versions of libbpf resulting in a SEGV whenever bpftrace is used.

Although `--help` and `-l` (list probe points) work OK that is because they do not actually use BPF! Anything that does use BPF will crash immediately,including `bpftrace --info` and simple one-liners such as:

    ~# bpftrace -e 't:raw_syscalls:sys_exit { printf("%s", comm); }'
    Attaching 1 probe...
    Segmentation fault

Fixes: https://github.com/iovisor/bpftrace/issues/2173
Fixes: bb3e56b06f9d ("bpftrace: Fix build with new libbpf")
    
Currently I have fixed this in two parts, firstly we make de-vendoring work by using a nasty hack and secondly we make the actual change!

Second part is (IMHO) "good" but I am concerned the first part is better handled elsewhere. The data structures I have introduced are copyed from v5.17 kernel headers and I not entirely clear whether or not they are candidates to backport into linux-libc-headers in kirkstone. In other words I wanted to document current status and ask for advice on how to land the needed changes (and I can move that to the mailing lists if needed now that I have something to point at ;-) ). If nothing else I hope the patch does, at least, clearly illustrate the problem.

### Contributor checklist

- [x] Changes have been tested
  + Fully tested on kirkstone and (AFAICT) all contributing components (linux-libc-headers, libbpf, bcc, bpftrace) have not changed in master.
- [x] `Signed-off-by` is present
- [x] The PR complies with the [Open Embedded Commit Patch Message Guidelines](http://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines)

### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)

### CODE REVIEW

File: dynamic-layers/openembedded-layer/recipes-devtools/bcc/bcc/0001-Vendor-just-enough-extra-headers-to-allow-libbpf-to-.patch [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+deliberate vendoring of some depreciated btf code that is being removed
+upstream because bcc was the only user! In other words the vendored code
+doesn't work the de-vendored libbpf because also ends up de-vendoring
+one of the Linux uapi headers.
+
+This is obviously an OE specific issue (due to the current combination
+of linux headers, libbpf and bcc). It's a bit of a mess and the right
+solution is probably to update the system UAPI headers but I am committing
+this for now simply so I can easily show why we must de-vendor libbpf in
+the first place!
```
Comment (kraj) [inline dynamic-layers/openembedded-layer/recipes-devtools/bcc/bcc/0001-Vendor-just-enough-extra-headers-to-allow-libbpf-to-.patch:18]: this patch has significant cost of maintenance in future if it  remains here, so I am a bit worried. Have you discussed this with upstream bcc community to see how they want to handle this situation for future ?

Comment (kraj) [reply dynamic-layers/openembedded-layer/recipes-devtools/bcc/bcc/0001-Vendor-just-enough-extra-headers-to-allow-libbpf-to-.patch:18]: I see, you mean linux UAPI headers right ? which we maybe doing soon,

Comment (daniel-thompson) [reply dynamic-layers/openembedded-layer/recipes-devtools/bcc/bcc/0001-Vendor-just-enough-extra-headers-to-allow-libbpf-to-.patch:18]: Yep, UAPI headers. If that gets updated then it should be easy to get bpftrace working on master by de-vendoring the bcc build.

I assume kirkstone will stay on the older UAPI though, right?

BTW and FYI, upstream discussion is here: https://github.com/iovisor/bcc/issues/4014

Comment (kraj) [reply dynamic-layers/openembedded-layer/recipes-devtools/bcc/bcc/0001-Vendor-just-enough-extra-headers-to-allow-libbpf-to-.patch:18]: Yes Kirk stone will stay where it is

Comment (b49020) [reply dynamic-layers/openembedded-layer/recipes-devtools/bcc/bcc/0001-Vendor-just-enough-extra-headers-to-allow-libbpf-to-.patch:18]: Hi @kraj,

Can you help backport this patch to kirkstone stable branch?

-Sumit

**Pull request merged**